src/server/services/stripe.service.ts

```diff
@@ -208,18 +209,37 @@ export const createBuzzSession = async ({
   user,
   returnUrl,
   priceId,
+  customAmount,
 }: Schema.CreateBuzzSessionInput & { customerId?: string; user: Schema.CreateCustomerInput }) => {
   const stripe = await getServerStripe();
 
   if (!customerId) {
     customerId = await createCustomer(user);
   }
 
+  const price = await dbRead.price.findUnique({
+    where: { id: priceId },
+    select: { productId: true, currency: true, type: true },
+  });
+  if (!price)
+    throw throwNotFoundError(`The product you are trying to purchase does not exists: ${priceId}`);
+
   const session = await stripe.checkout.sessions.create({
     customer: customerId,
     cancel_url: returnUrl,
-    line_items: [{ price: priceId, quantity: 1 }],
-    mode: 'payment',
+    line_items: [
+      customAmount
+        ? {
+            price_data: {
+              unit_amount: customAmount * 100,
```

Comment:
we should not be doing this, customAmount should come multiplied by 100 already (i.e, 5.99 is 599)